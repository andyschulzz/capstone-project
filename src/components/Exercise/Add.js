import React, { useState } from 'react'
import { useForm } from 'react-hook-form'
import Form from './Form'
import ButtonBackToHome from '../common/ButtonBackToHome'
import ButtonEdit from './ButtonEdit'
import ButtonSave from './ButtonSave'
import * as S from './Form.styles'
import PropTypes from 'prop-types'

Add.propTypes = {
  handleExerciseAdd: PropTypes.func.isRequired,
}

export default function Add({ handleExerciseAdd }) {
  const [disabled, setDisabled] = useState(false)
  const { register, handleSubmit, formState } = useForm({ mode: 'onChange' })

  return (
    <>
      <S.ButtonWrapper>
        <ButtonBackToHome />
        {(!disabled && <ButtonSave formState={formState} form={'add'} />) || (
          <ButtonEdit />
        )}
      </S.ButtonWrapper>
      <S.Wrapper id="add" onSubmit={handleSubmit(handleAdd)}>
        <Form register={register} disabled={disabled} />
      </S.Wrapper>
    </>
  )
  function handleAdd(data) {
    Object.keys(data).map(key => (data[key] = data[key].trim()))
    setDisabled(!disabled)
    if (!disabled) {
      handleExerciseAdd(data.name, data.type, data.instructions)
    }
  }
}
