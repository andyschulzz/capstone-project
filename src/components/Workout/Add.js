import React from 'react'
import ExerciseList from './ExerciseList'
import ButtonSave from './ButtonSave'
import { useHistory } from 'react-router-dom'
import * as S from './Add.styles'
import PropTypes from 'prop-types'
import { useForm } from 'react-hook-form'

Add.propTypes = {
  exercises: PropTypes.array.isRequired,
  handleWorkoutAdd: PropTypes.func.isRequired,
  selectedWorkouts: PropTypes.array.isRequired,
}

export default function Add({
  exercises,
  handleWorkoutAdd,
  handleWorkoutTitle,
  selectedWorkouts,
}) {
  const history = useHistory()
  const { register, handleSubmit, formState } = useForm({
    mode: 'onChange',
  })

  return (
    <>
      <S.ButtonWrapper>
        <h3>Build your Routine</h3>
        {selectedWorkouts.length === 0 ? (
          <ButtonSave form={'add'} formState={false} />
        ) : (
          <ButtonSave form={'add'} formState={formState} />
        )}
      </S.ButtonWrapper>
      <form id="add" onSubmit={handleSubmit(onSubmit)}>
        <S.Textarea
          ref={register({ required: true })}
          autoFocus
          type="text"
          name="name"
          id="name"
          placeholder="Name your workout"
        />
      </form>
      <S.Wrapper>
        <ExerciseList
          exercises={exercises}
          handleWorkoutAdd={handleWorkoutAdd}
        />
      </S.Wrapper>
    </>
  )

  function onSubmit(input) {
    handleWorkoutTitle(input.name)
    history.push('/workouts/details')
  }
}
