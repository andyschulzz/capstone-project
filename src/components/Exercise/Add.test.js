import React from 'react'
import { render, cleanup } from '@testing-library/react'
import Add from './Add.js'
import { MemoryRouter as Router } from 'react-router-dom'

afterEach(cleanup)

const handleExerciseAddMock = jest.fn()
describe('<Add />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <Router>
        <Add handleExerciseAdd={handleExerciseAddMock} />
      </Router>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should be empty', () => {
    const { getByPlaceholderText } = render(
      <Router>
        <Add handleExerciseAdd={handleExerciseAddMock} />
      </Router>
    )
    const inputNode = getByPlaceholderText('Name of the exercise?')
    expect(inputNode).toBeVisible()
  })
})
