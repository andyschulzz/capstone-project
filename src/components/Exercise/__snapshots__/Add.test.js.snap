// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Add /> should render correctly 1`] = `
<DocumentFragment>
  .c0 {
  overflow-y: auto;
}

.c0:valid button[class*='Green'] {
  background: hsl(120,50%,50%);
}

.c4 {
  display: grid;
  max-height: 100%;
  gap: 10px;
  padding: 0 40px 30px;
  position: relative;
}

.c4 img {
  width: 130px;
  height: 130px;
  position: absolute;
  right: 48px;
  top: 64px;
}

.c4 p {
  margin: 0;
  font-size: 10px;
  color: #ccc;
}

.c6 {
  font-weight: bold;
  font-size: 1.1rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: none;
  outline: none;
  padding: 0px;
  margin-bottom: 10px;
  width: 100%;
  height: 40px;
  resize: none;
  font-size: 1.7rem;
  font-weight: bold;
  background-color: #fff;
  overflow-y: hidden;
  color: #333;
}

.c5::-webkit-input-placeholder {
  font-weight: 300;
}

.c5::-moz-placeholder {
  font-weight: 300;
}

.c5:-ms-input-placeholder {
  font-weight: 300;
}

.c5::placeholder {
  font-weight: 300;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: none;
  outline: none;
  padding: 10px;
  margin-bottom: 10px;
  width: 100%;
  height: 40px;
  resize: none;
  font-size: 1rem;
  font-weight: 400;
  background-color: #fff;
  overflow-y: hidden;
  color: #333;
}

.c7::-webkit-input-placeholder {
  font-weight: 300;
}

.c7::-moz-placeholder {
  font-weight: 300;
}

.c7:-ms-input-placeholder {
  font-weight: 300;
}

.c7::placeholder {
  font-weight: 300;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: none;
  outline: none;
  padding: 10px;
  margin-bottom: 10px;
  width: 100%;
  height: 400px;
  resize: none;
  font-size: 1rem;
  font-weight: 400;
  background-color: #fff;
  overflow-y: auto;
  color: #333;
}

.c8::-webkit-input-placeholder {
  font-weight: 300;
}

.c8::-moz-placeholder {
  font-weight: 300;
}

.c8:-ms-input-placeholder {
  font-weight: 300;
}

.c8::placeholder {
  font-weight: 300;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 40px 40px 30px;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 1;
  background: #fff;
}

.c2 {
  width: 25px;
  height: 25px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: none;
  border: none;
  outline: none;
  border-radius: 5px;
  margin-left: auto;
  padding: 5px 15px;
  font-size: 1rem;
  height: 36px;
  cursor: pointer;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: hsla(120,50%,50%,0.2);
}

.c3 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: white;
}

<form
    class="c0"
    data-testid="add-form"
    id="add"
  >
    <div
      class="c1"
    >
      <a
        href="/"
      >
        <img
          alt=""
          class="c2"
          src="back.png"
        />
      </a>
      <button
        class="Button-sc-18sujpt-0 c3"
      >
        Save
      </button>
    </div>
    <div
      class="c4"
    >
      <textarea
        class="c5"
        id="name"
        maxlength="50"
        minlength="4"
        name="name"
        placeholder="Name of the exercise?"
        required=""
        type="text"
      />
      <label
        class="c6"
        for="type"
      >
        Type
      </label>
      <textarea
        class="c7"
        id="type"
        maxlength="30"
        minlength="4"
        name="type"
        placeholder="Type of the exercise?"
        required=""
        type="text"
      />
      <label
        class="c6"
        for="instructions"
      >
        Instructions
      </label>
      <textarea
        class="c8"
        id="instructions"
        minlength="4"
        name="instructions"
        placeholder="Please explain the exercise!"
        required=""
        type="text"
      />
    </div>
  </form>
</DocumentFragment>
`;
